webapp/src/webapp/connections/views/create_update_connection/connection_type_form.cljs

```diff
@@ -0,0 +1,120 @@
+            [clojure.string :as str]
+            [reagent.core :as r]
+            [webapp.connections.utilities :as utils]))
+
+(def connections-type
+  [{:icon (r/as-element [:> Database {:size 16}])
+    :title "Database"
+    :subtitle "For PostgreSQL, MySQL, Microsoft SQL and more."
+    :value "database"}
+   {:icon (r/as-element [:> SquareTerminal {:size 16}])
+    :title "Shell"
+    :subtitle "Custom connection for your services."
+    :value "custom"}
+   {:icon (r/as-element [:> SquareTerminal {:size 16}])
+    :title "SSH"
+    :subtitle "Secure shell protocol for remote access."
+    :value "ssh"}
+   {:icon (r/as-element [:> Workflow {:size 16}])
+    :title "TCP"
+    :subtitle "Transmission protocol for reliable transmission of data."
+    :value "tcp"}
+   {:icon (r/as-element [:> AppWindow {:size 16}])
+    :title "Application"
+    :subtitle "For Ruby on Rails, Python, Node JS and more."
+    :value "application"}])
+
+(def connections-subtypes
+  {"database" [{:value :postgres :title "PostgreSQL"}
+               {:value :mysql :title "MySQL"}
+               {:value :mssql :title "Microsoft SQL"}
+               {:value :oracledb :title "Oracle DB"}
+               {:value :mongodb :title "MongoDB"}]
+   "custom" []
+   "application" [{:value :ruby-on-rails :title "Ruby on Rails"}
+                  {:value :python :title "Python"}
+                  {:value :nodejs :title "Node JS"}
+                  {:value :clojure :title "Clojure"}]})
+
+(defn set-connection-type-context [value connection-type connection-subtype config-file-name]
+  (cond
+    (= value "database") (do (reset! connection-type "database")
+                             (reset! connection-subtype nil))
+
+    (= value "custom") (do (reset! connection-type "custom")
+                           (reset! connection-subtype nil))
+
+    (= value "ssh") (do (reset! connection-type "custom")
+                        (reset! connection-subtype "ssh")
+                        (reset! config-file-name "SSH_PRIVATE_KEY"))
+
+    (= value "tcp") (do (reset! connection-type "application")
+                        (reset! connection-subtype "tcp"))
+
+    (= value "application") (do (reset! connection-type "application")
+                                (reset! connection-subtype nil))))
+
+(defn is-connection-type-selected [value connection-type connection-subtype]
+  (cond
+    (= value "database") (= value connection-type)
+
+    (= value "custom") (and (= value connection-type)
+                            (not= connection-subtype "ssh"))
+
+    (= value "ssh") (and (= value connection-subtype)
+                         (= connection-type "custom"))
+
+    (= value "tcp") (and (= value connection-subtype)
+                         (= connection-type "application"))
+
+    (= value "application") (and (= value connection-type)
+                                 (not= connection-subtype "tcp"))))
+
```

Comment:
I understand the need of this in here, but it makes very complicated to understand. Is it because you have to manage some information for the form submit? If so, why not make it simpler here using only the value to set the styles and let the submit worry about these details?

Reply:
It's not. It isn't very easy because, SSH and TCP are not a type, but we are using them as a type in the UI and UX.
In this case, I'm using it to tell me which card of connection type is selected.
So I'm comparing the value selected with the atoms. I would like to compare only `@connection-type (TCP) = value (TCP)` but it's not true

Reply:
We will change how the form works today. We will send the management state to reframe. 
But it's a big change in the logic of this feature so we will do it in the new PR.